### ISSUE TITLE
Implicit stringify converter from log methods

### ISSUE THREAD

**Issue by cornerman** [outr/scribe#51]
Just stumbled on this error in my code which confused me:

```
[error] /home/cornerman/projects/wust2/util/shared/src/main/scala/package.scala:19:21: No implicit view available from Throwable => String.
[error]         scribe.error(e)
```

So, the log methods in [`LoggerSupport`](https://github.com/outr/scribe/blob/master/core/shared/src/main/scala/scribe/LoggerSupport.scala) take an implicit converter from the given type to `String`. But I am not sure what the use case is: Let's say I would want to log something else than a String. Then I would need to put an implicit function `Something => String` in my scope, which I would not want to have as it is to generic. Or I would need to supply it explicitly which is kind of tedious.

How about using a typeclass `Loggable` instead of an implicit view? So we could define certain types as loggable and just pass them to the appropriate log methods, which would feel much better to have as an implicit.

```
trait Loggable[T] {
  def logString(t: T): String
}
```

Furthermore, we could have overloads just expecting a String, to not have the typeclass overhead on every logline.

**Comment by darkfrog26** [comment]
The purpose behind this is to support original types being able to be received by the logging system.  For example, if I'm writing a database logger and want to provide a structured logging of specific types I can do so in Scribe.  For most people, they always just pass Strings, and in your case, you can use:

`import scribe.LogRecord.Stringify._`

That will give you implicit conversion support for `Throwable`.  I need to write an update to auto-handle that though.

I appreciate the feedback, and it's a rather new feature that still needs to be iterated upon, but the goal is the capacity to access more than just a `String` when desired for more complex logging.

**Comment by darkfrog26** [comment]
PR merged, closing.

**Comment by darkfrog26** [comment]
Released 2.3.2 with these changes
